ego/cli/bundle.go

```diff
@@ -0,0 +1,240 @@
+		}
+	}
+
+	if isAlreadyBundled {
+		return ErrFileIsAlreadyBundled
+	}
+
+	if !isEnclave {
+		return ErrFileIsNotAnEGoBinary
+	}
+
+	if !isGoBinary {
+		return ErrFileIsNotAnEGoBinary
+	}
+
+	// Check if the binary has been signed with `ego sign` yet by checking if the payload has been appended.
+	// We do not check here if it is actually valid. If it is not valid, the binary can still be bundled, but during launch the enclave will refuse to run.
+	payloadSize, _, _, err := getPayloadInformation(file)
+	if err != nil {
+		return err
+	}
+
+	if payloadSize == 0 {
+		return ErrFileHasNotBeenSignedYet
+	}
+
+	return nil
+}
+
+// buildImage builds a gzip-compressed tarball containing the EGo runtime.
+func (c *Cli) buildImage(enclaveFilename string) (string, error) {
+	// Copy the runtime image to the tarPreparedContent
+	egoHostPath := c.getEgoHostPath()
+	egoEnclavePath := filepath.Join(c.egoPath, "share", "ego-enclave")
+
+	// Create the tar file
+	tempFile, err := c.fs.TempFile("", "ego-runtime")
+	if err != nil {
+		return "", err
+	}
+	defer tempFile.Close()
+
+	// Setup tar writer with gzip compression
+	compressedWriter := gzip.NewWriter(tempFile)
+	defer compressedWriter.Close()
+	tarWriter := tar.NewWriter(compressedWriter)
+	defer tarWriter.Close()
+
+	// Add the runtime image to the tar file
+	egoHostTargetPath := filepath.Join("bin", "ego-host")
+	if err := c.addToArchive(tarWriter, egoHostPath, egoHostTargetPath); err != nil {
+		return "", err
+	}
+	egoEnclaveTargetPath := filepath.Join("share", "ego-enclave")
+	if err := c.addToArchive(tarWriter, egoEnclavePath, egoEnclaveTargetPath); err != nil {
+		return "", err
+	}
+
+	// Add the enclave to the tar file
+	if err := c.addToArchive(tarWriter, enclaveFilename, "enclave"); err != nil {
+		return "", err
+	}
+
+	return tempFile.Name(), err
+}
+
+// addToArchive adds a file from the filesystem to the tar archive.
+func (c *Cli) addToArchive(tw *tar.Writer, sourceFilename string, targetFilename string) error {
+	file, err := c.fs.Open(sourceFilename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	sourceFileInfo, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	header, err := tar.FileInfoHeader(sourceFileInfo, sourceFileInfo.Name())
+	if err != nil {
+		return err
+	}
+
+	header.Name = targetFilename
+
+	if err := tw.WriteHeader(header); err != nil {
+		return err
+	}
+
+	n, err := io.Copy(tw, file)
+	if err != nil {
+		return err
+	}
+
+	if n != sourceFileInfo.Size() {
+		return fmt.Errorf("file size mismatch: %d written instead of expected %d bytes", n, sourceFileInfo.Size())
+	}
+
+	return nil
+}
+
+// prepareBundle copies the "empty" bundle executable to the target location and names it after the enclave to be bundled.
+func (c *Cli) prepareBundle(filename string) (string, error) {
+	// Create new file which eventually ends up as the packed binary
+	filenameBase := filepath.Base(filename)
+	filenameBaseNoExt := strings.Split(filenameBase, ".")[0]
```

Comment:
I think the command should better be `ego bundle <in> <out>`

Reply:
I would maybe do both: use the output when specified, but also allow "just" ego bundle <in> similar as one can just call "ego-go build". 

(Though that splitting might need a little refinement, given that it can do weird things when the file starts with a dot).